diff --git a/libtiff/tif_dir.c b/libtiff/tif_dir.c
index a6c254fc..d932e412 100644
--- a/libtiff/tif_dir.c
+++ b/libtiff/tif_dir.c
@@ -29,6 +29,7 @@
  * (and also some miscellaneous stuff)
  */
 #include "tiffiop.h"
+#include <stdio.h>
 #include <float.h>	/*--: for Rational2Double */
 
 /*
@@ -1665,6 +1666,7 @@ TIFFNumberOfDirectories(TIFF* tif)
 int
 TIFFSetDirectory(TIFF* tif, uint16_t dirn)
 {
+	printf("jiezhu");
 	uint64_t nextdir;
 	uint16_t n;
 
diff --git a/libtiff/tif_dirread.c b/libtiff/tif_dirread.c
index d84147a0..43e068fb 100644
--- a/libtiff/tif_dirread.c
+++ b/libtiff/tif_dirread.c
@@ -37,6 +37,7 @@
 #include "tiffconf.h"
 #include "tiffiop.h"
 #include <float.h>
+#include <stdio.h>
 #include <stdlib.h>
 
 #define FAILED_FII    ((uint32_t) -1)
@@ -5909,7 +5910,7 @@ static void allocChoppedUpStripArrays(TIFF* tif, uint32_t nstrips,
  */
 static void
 ChopUpSingleUncompressedStrip(TIFF* tif)
-{
+{	
 	register TIFFDirectory *td = &tif->tif_dir;
 	uint64_t bytecount;
 	uint64_t offset;
@@ -5960,6 +5961,7 @@ ChopUpSingleUncompressedStrip(TIFF* tif)
 
         /* If we are going to allocate a lot of memory, make sure that the */
         /* file is as big as needed */
+#ifdef MAGMA_ENABLE_FIXES
         if( tif->tif_mode == O_RDONLY &&
             nstrips > 1000000 &&
             (offset >= TIFFGetFileSize(tif) ||
@@ -5967,6 +5969,13 @@ ChopUpSingleUncompressedStrip(TIFF* tif)
         {
             return;
         }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+        MAGMA_LOG("%MAGMA_BUG%", MAGMA_AND(tif->tif_mode == O_RDONLY, \
+	        MAGMA_AND(nstrips > 1000000, \
+	        MAGMA_OR(offset >= TIFFGetFileSize(tif), \
+	         stripbytes * (nstrips - 1) > (TIFFGetFileSize(tif) - offset)))));
+#endif
 
         allocChoppedUpStripArrays(tif, nstrips, stripbytes, rowsperstrip);
 }
