diff --git a/src/sndfile.c b/src/sndfile.c
index 020e0987..471f8d02 100644
--- a/src/sndfile.c
+++ b/src/sndfile.c
@@ -3126,6 +3126,11 @@ psf_open_file (SF_PRIVATE *psf, SF_INFO *sfinfo)
 			psf->sf.format = format_from_extension (psf) ;
 		} ;
 
+#ifdef MAGMA_ENABLE_CANARIES
+	if (psf->sf.format == SF_FORMAT_FLAC) {
+		MAGMA_LOG("flac", psf->sf.format == SF_FORMAT_FLAC);
+	}
+#endif
 	/* Prevent unnecessary seeks */
 	psf->last_op = psf->file.mode ;
 
