diff --git a/src/sndfile.c b/src/sndfile.c
index 365c1361..2d7431cf 100644
--- a/src/sndfile.c
+++ b/src/sndfile.c
@@ -3114,6 +3114,13 @@ psf_open_file (SF_PRIVATE *psf, SF_INFO *sfinfo)
 			psf->sf.format = format_from_extension (psf) ;
 		} ;
 
+#ifdef MAGMA_ENABLE_CANARIES
+	if (SF_CONTAINER (psf->sf.format) == SF_FORMAT_PAF) {
+		MAGMA_LOG("paf", SF_CONTAINER (psf->sf.format) == SF_FORMAT_PAF);
+	} else {
+		MAGMA_EXIT;
+	}
+#endif
 	/* Prevent unnecessary seeks */
 	psf->last_op = psf->file.mode ;
 
